code/game/objects/items.dm

```diff
@@ -2048,3 +1984,38 @@
 			BARE WOUND: <font size='1'><a href='byond://?_src_=vars;[HrefToken()];item_to_tweak=[REF(src)];var_tweak=bare wound' id='bare wound'>[bare_wound_bonus]</a>
 		</font>
 	"}
+
+/// Fetches, or lazyloads, our embedding datum
+/obj/item/proc/get_embed()
+	RETURN_TYPE(/datum/embedding)
+	if (embed_data)
+		return embed_data
+	if (embed_type)
+		embed_data = new embed_type(src)
+	return embed_data
```

Comment:
Something notable about this you might want to address:

If something calls `get_embed` on a qdeleted item, it will re-instantiate any embed type WHILE the object is deleting itself (this is an issue we've had/have with language holders) which can cause hard deletes (as you make a new datum)

Might want to harden it with a `if(!QDELTED)` check